controllers/positions.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs')
+const Positions = require('../data/positions.json')
+
+const createPosition = (req, res) => {
+  const newPosition = {
+      id: new Date().getTime().toString(),
+      idClient: req.query.idClient,
+      jobDescription: req.query.jobDescription,
+      vacancy: req.query.vacancy,
+      professionalProfiles: req.query.professionalProfiles,
+      status: req.query.status
+    }
+    Positions.push(newPosition)
+    fs.writeFile('./data/positions.json', JSON.stringify(Positions), {}, err => {
+      if (err) {
+        res.status(400).send(err)
+      } else {
+        res.status(201).json(newPosition)
+      }
+    })
+}
+
+const updatePosition = (req, res) => {
+  let updatedPosition
+  const updatedPositions =  Positions.map((position) => {
+    if(position.id === req.query.id) {
```

Comment:
What happens if the id does not exist on the list?
The list should not be updated and the endpoint should return with a 404 error.

Reply:
I implemented a flag = 1 when the if statement is entered.
` if(flag == 0) res.status(404).send('position not found')` 
`else if (JSON.stringify(updatedPositions) == JSON.stringify(Positions)) res.status(200).send('position found but no changes were made')` //here if the position was found be the mapped array is equal to the original file, no changes made

Reply:
Ok, you can use the `updatedPositions` as a flag, if it is empty means any positions was not found.

Reply:
fixed